commit 5bd3cadc732a7d245cbe3a1bfd00e4461d59a2b1 (refs/changes/50/1650/3)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2017-03-17T18:49:59+00:00 (2 years, 7 months ago)
    
    joyent/node-mname#17 crash when receiving unknown query types
    Reviewed by: Brittany Wald <brittany.wald@joyent.com>

diff --git a/lib/server.js b/lib/server.js
index 1871b6c..e22e1a8 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -178,10 +178,16 @@ Server.prototype.listenTcp = function listenTcp(opts, callback) {
                         try {
                                 query = Query.parse(qopts);
                                 qs[query.id] = true;
+                                /*
+                                 * Note that it's important we call the
+                                 * query.operation() function under the try{}
+                                 * block here (it can throw).
+                                 */
                                 log = log.child({
                                         qId: query.id,
                                         qName: query.name(),
-                                        qType: query.type()
+                                        qType: query.type(),
+                                        qOp: query.operation()
                                 });
                                 query._log = log;
                         } catch (e) {
@@ -306,10 +312,16 @@ Server.prototype.listenUdp = function listenUdp(opts, callback) {
 
                 try {
                         query = Query.parse(qopts);
+                        /*
+                         * Note that it's important we call the
+                         * query.operation() function under the try{}
+                         * block here (it can throw).
+                         */
                         log = log.child({
                                 qId: query.id,
                                 qName: query.name(),
-                                qType: query.type()
+                                qType: query.type(),
+                                qOp: query.operation()
                         });
                         query._log = log;
                 } catch (e) {
diff --git a/package.json b/package.json
index f7386af..46f6721 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "mname",
   "description": "DNS server library for node.js",
-  "version": "1.3.4",
+  "version": "1.3.5",
   "author": "arekinath <alex@cooperi.net>",
   "contributors": [
     "Mark Cavage",
